android/demo/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -236,37 +212,36 @@ public void onServiceDisconnected(ComponentName arg0) {
         @Override
         public void onReceive(Context context, Intent intent) {
             if (intent != null) {
-                final List<String> validUrls = intent.getStringArrayListExtra(ENVOY_DATA_URLS_SUCCEEDED);
-                if (validUrls != null && !validUrls.isEmpty()) {
-                    String envoyUrl = validUrls.get(0);
-                    Log.i(TAG, "Received valid urls: " + TextUtils.join(", ", validUrls));
-                    // Select the fastest one
-                    CronetNetworking.initializeCronetEngine(context, envoyUrl); // reInitializeIfNeeded set to false
-                    new Thread() {
-                        @Override
-                        public void run() {
-                            try {
-                                URL url = new URL("https://api.ipify.org/");
-                                BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream()));
-                                String inputLine;
-                                StringBuilder response = new StringBuilder();
-                                while ((inputLine = in.readLine()) != null)
-                                    response.append(inputLine);
-                                in.close();
-
-                                Log.d(TAG, "proxied request returns " + response.toString());
-                            } catch (MalformedURLException e) {
-                                Log.e(TAG, "failed to proxy request ", e);
-                            } catch (IOException e) {
-                                Log.e(TAG, "failed to read response ", e);
+                final String envoyUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED);
+                if (envoyUrl != null && !envoyUrl.isEmpty()) {
+                    Log.i(TAG, "Received valid url: " + envoyUrl);
+                    if (CronetNetworking.cronetEngine() != null) {
+                        Log.i(TAG, "Cronet already started");
+                    } else {
+                        CronetNetworking.initializeCronetEngine(context, envoyUrl); // reInitializeIfNeeded set to false
+                        new Thread() {
+                            @Override
+                            public void run() {
+                                try {
+                                    URL url = new URL("https://api.ipify.org/");
```

Comment:
What is this url?

Reply:
it's an API that returns your public IP... the stuff in `demo` is super old and hasn't really been touched in years (I think @mnbogner fixed some build issues in there, but not much else)